### PR TITLE
Document custom assets

### PR DESCRIPTION
**Pull request by pablobm**
We are not documenting how to add custom assets to complement or override Administrate's own. In fact I recently came across a client project where this was done the wrong way, causing breakage.

So here's some documentation on it, as well as adds examples to the example app. The examples (particularly the CSS one) are a bit artificial, but I couldn't think of anything else. Suggestions welcome.

### CODE REVIEW

File: docs/customizing_page_views.md
Comment (elias19r) [inline docs/customizing_page_views.md:118]: Looks good, thank you!

@pablobm @nickcharlton I wonder what we should to do with the following generators:
- `lib/generators/administrate/assets/assets_generator.rb`
- `lib/generators/administrate/assets/javascripts_generator.rb`
- `lib/generators/administrate/assets/stylesheets_generator.rb`

They are another way of customizing assets. As far as I understand, by running those generators, Administrate's non-compiled assets are copied to the users' app. So users can edit a non-compiled asset and _compile it themselves_ if their app is set up for that. And then their assets will take precedence over the ones from the Administrate gem.

I lean toward removing those generators and making users customize assets by appending to Administrate's CSS/JS, just like the example in this PR, instead of completely replacing them with their own.

**Review by elias19r — APPROVED** [review]

File: spec/example_app/app/assets/stylesheets/admin.css
Comment (elias19r) [reply spec/example_app/app/assets/stylesheets/admin.css:1]: Hound's got a point. Should this be

```css
/*
*= require_tree ./admin
*/
```

Comment (pablobm) [reply spec/example_app/app/assets/stylesheets/admin.css:1]: Good point. To be honest, as long as it works and Hound shuts up, I'm in :+1:

File: docs/customizing_page_views.md
Comment (pablobm) [reply docs/customizing_page_views.md:118]: I agree. I think that this way of customising assets is more likely to cause issues than to solve them. Adding additional, separate assets to add new behaviour (or override CSS) feels like a better solution to me. Then if anyone wants to fully override Administrate's own assets, they can copy them over manually and do that.

Comment (nickcharlton) [reply docs/customizing_page_views.md:118]: Yeah, that makes sense to me 👍

**Review by nickcharlton — APPROVED** [review]
I think it'd be good to remove those generators, but in another PR (I'll get around to it eventually, but if someone else wanted to, go ahead!). And then we can merge this once #2397 goes in.

**Comment by pablobm** [comment]
Forgot about this one after the merge of #2397. I have rebased and made a change (converted SCSS to CSS). Nor merging just yet as I'm not sure if it'll need to wait for https://github.com/thoughtbot/administrate/pull/2673 now.

**Comment by pablobm** [comment]
And related PR to remove old asset generators at https://github.com/thoughtbot/administrate/pull/2674, as discussed above.

**Review by Blackskar875 — APPROVED** [review]

**Pull request merged**
